README.md

```diff
@@ -34,6 +34,85 @@ ERROR: SystemError: opening file foo: No such file or directory
  in open at iostream.jl:99
 ```
 
+### @mendable
+
+Some functions you want to mock are called from the function you are testing. If the function you are testing is compiled, you will not be able to mock the internal function. That is where `@mendable` can help, it will allow you to mock the internal functions to your needs. As show in this example.
+
+```
+julia>  function d2()
+            a
+        end
+d2 (generic function with 1 method)
+
+julia>  function d()
+            d2()
+        end
+d (generic function with 1 method)
+
+julia> test() = @mendable d2()
+test (generic function with 1 method)
+
+julia> test()
+ERROR: UndefVarError: a not defined
+ in d2 at none:2
+ in test at none:1
+ in eval at ./boot.jl:264
+
+julia>  function d2()
+            b = 2
+        end
+d2 (generic function with 1 method)
+
+julia> test()
+2
+```
+
+### Patch
+
+`Patch` allows similar behaviour as you would get from using `mend` directly.
+
+```
+julia> replacement = (name) -> name == "foo" ? IOBuffer("bar") : Original.open(name)
+(anonymous function)
+
+julia> patch = Patch(open, replacement, (AbstractString,))
+Mocking.Patch(open,(anonymous function),Mocking.Signature(Type[AbstractString]))
+
+julia> mend(patch) do
+           readall(open("foo"))
+       end
+"bar"
+```
+
+We can also define multiple patches to make the code more readable.
```

Comment:
Change this to use an array of patches which is what I think the standard use case will be.